src/utils/get-import-flavor-of-node.ts

```diff
@@ -0,0 +1,29 @@
+import {
+    importFlavorIgnore,
+    importFlavorRegular,
+    importFlavorSideEffect,
+    importFlavorType,
+} from '../constants';
+import type { GetImportFlavorOfNode } from '../types';
+
+/**
+ * Classifies nodes by import-flavor, primarily informing whether the node is a candidate for merging
+ *
+ * @param node
+ * @returns {("prettier-ignore"|"regular"|"side-effect"|"type")}
```

Comment:
Using jsdoc together with typescript is a bit tricky.  I'd say this belongs as a return type in typescript on the function below, and we should use `@returns` to describe the return with a short phrase or sentence.  In this case, we could say `@returns the flavor of the import node`, or something like that.